src/class-parsely.php

```diff
@@ -653,4 +670,93 @@ private function are_credentials_managed(): bool {
 
 		return true;
 	}
+
+	/**
+	 * Sets the values of managed options.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 */
+	private function set_managed_options(): void {
+		$managed_options = apply_filters( 'wp_parsely_managed_options', false );
+
+		if ( ! is_array( $managed_options ) ) {
+			return;
+		}
+
+		// Don't allow certain options to be set as managed.
+		unset(
+			$managed_options['apikey'],
+			$managed_options['api_secret'],
+			$managed_options['metadata_secret'],
+			$managed_options['track_post_types'],
+			$managed_options['track_page_types'],
+			$managed_options['plugin_version']
+		);
+
+		/**
+		 * Current options.
+		 *
+		 * @var Parsely_Options $current_options
+		 */
+		$current_options = get_option( self::OPTIONS_KEY, array() );
+
+		// Set managed options values.
+		foreach ( $managed_options as $key => $value ) {
+			$is_option_valid = isset( $this->option_defaults[ $key ] );
+
+			if ( $is_option_valid ) {
+				if ( null === $value ) {
+					// When null, the option gets its value from the database.
+					$this->managed_options[ $key ] =
+						$current_options[ $key ] ?? $this->option_defaults[ $key ];
+				} else {
+					$this->managed_options[ $key ] =
+						$this->sanitize_managed_option( $key, $value );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Sanitizes the value of the passed managed option.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 *
+	 * @param string      $option_id The option's ID.
+	 * @param bool|string $value The option's value.
+	 *
+	 * @return bool|string The sanitized option value.
+	 */
+	private function sanitize_managed_option( string $option_id, $value ) {
+		$option_value_type = gettype( $this->option_defaults[ $option_id ] );
+
+		if ( 'boolean' === $option_value_type && ! is_bool( $value ) ) {
+			return false;
+		}
+
+		if ( 'string' === $option_value_type ) {
+			if ( ! is_string( $value ) ) {
+				$value = strval( $value );
+			}
+
+			// String options that are restricted to specific values.
+			$restricted_value_options = array(
+				'custom_taxonomy_section' => Settings_Page::get_section_taxonomies(),
```

Comment:
Minor: This feels a bit risky. Even though it's calling code within our own plugin, I worry that it may cause problems at some point due to load order or something else.

I'm not sure if we need to enforce that an incorrect taxonomy was passed in, but I think it's also okay to leave as-is.

Reply:
I get your point.

- We could move this to the `Validator` class, but as long as we're moving this to an external file, we've got the same load order dangers.
- We could implement the function within this file, but I think this function is a bit out of scope for this class.

I think that validating values makes sense, and I think I'd leave this as-is for the moment, until we can come up with something better.